packages/components/src/FormField/FormField.tsx

```diff
@@ -3,219 +3,195 @@ import React, {
   FocusEvent,
   KeyboardEvent,
   MutableRefObject,
+  PropsWithChildren,
   useEffect,
-  useImperativeHandle,
   useState,
 } from "react";
 import classnames from "classnames";
-import uuid from "uuid";
-import { Controller, useForm, useFormContext } from "react-hook-form";
 import styles from "./FormField.css";
 import { FormFieldProps } from "./FormFieldTypes";
 import { FormLabel } from "./FormLabel";
 import { FieldWrapper } from "./FieldWrapper";
 import { FormSpinner } from "./FormSpinner";
+import { useFormController } from "./useFormController";
 import { Icon } from "../Icon";
 import { InputValidation } from "../InputValidation";
 
-export function FormField(props: FormFieldProps) {
+/**
+ * Disabling max line statements here as this component
+ * has a lot of handlers and there is no way around this.
+ */
+// eslint-disable-next-line max-statements
+export function FormField(props: PropsWithChildren<FormFieldProps>) {
   const {
-    actionsRef,
     autocomplete = true,
     children,
     defaultValue,
     disabled,
     inline,
-    inputRef,
+    name,
     keyboard,
     loading,
-    max,
     maxLength,
+    max,
     min,
-    name,
     readonly,
     rows,
     type = "text",
-    validations,
     value,
+    validations,
+    inputRef,
     onChange,
+    onValidation,
     onEnter,
-    onFocus,
     onBlur,
-    onValidation,
+    onFocus,
   } = props;
 
-  const { control, errors, setValue, watch } =
-    useFormContext() != undefined
-      ? useFormContext()
-      : useForm({ mode: "onTouched" });
-
-  const [identifier] = useState(uuid.v1());
-
-  /**
-   * Generate a name if one is not supplied, this is the name
-   * that will be used for react-hook-form and not neccessarily
-   * attached to the DOM
-   */
-  const [controlledName] = useState(
-    name ? name : `generatedName--${identifier}`,
-  );
+  const { error, field } = useFormController({
+    name,
+    value: value || defaultValue,
```

Comment:
any reason to use `||` over `??` ? If for some reason you wanted a value of 0/false/etc, the default would be used instead.

Reply:
This has been changed.